pkg/image/api/types.go

```diff
@@ -60,6 +77,41 @@ type ImageLayer struct {
 	Size int64
 }
 
+// SignatureValidationResult holds a result of signature validation.
+type SignatureValidationResult string
+
+// ImageSignature holds a signature of an image.
+type ImageSignature struct {
+	// Describes a type of stored blob. Provided by caller.
+	Type string
+	// An opaque binary string which is an image's signature.
+	Content []byte
+	// A time of last server-side validation.
+	ValidatedOn unversioned.Time
+	// Result of validation performed at ValidatedOn.
+	ValidationResult SignatureValidationResult
+
+	// At most one of the following pointers can point to details. If the signature type isn't recognized,
+	// they will all be nil.
+
+	// Details for signature of type "atomic".
+	AtomicSignatureDetails *AtomicSignatureDetails
```

Comment:
> As for the version. If we find the `atomic` kind unsufficient in the future, we can either change the name completely or append a version suffix if the next format will be similar enough. The combination of `Type` and `Version` should map to a particular signature details structure and to a particular parsing function. I don't find it necessary to have two fields for this purpose instead of one. But I don't have a strong opinion on this one. @mtrmac what's your opinion on a version field inside the signature?

It is not obvious to me that “signature type” and “signature version” are distinct concepts. In terms of interoperability, “new major version” and “new type” mean the same thing: all users need to be updated.

So, the signature blob contains a single `"type": "atomic container signature"` marker, but it also anticipates ways to extend the format without having to change the `"type"`: adding a new field to `"critical"` inside the signature will prevent clients who don’t understand it from accepting the signature.

I don’t know how that maps into the k8s semantics and conventions WRT unexpected / missing fields. (OpenShift might need to add versioning to `AtomicSignatureDetails` even if the actual signature does not need version numbers.)